Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -130,11 +136,107 @@ fun IncludeCertificateQrcodeCardBinding.bindValidityViews(
             startValidationCheckButton.isVisible = isPersonDetails
         }
     }
+}
+
+@Suppress("LongParameterList", "ComplexMethod")
+fun PersonOverviewItemBinding.setUIState(
+    primaryCertificate: CwaCovidCertificate,
+    secondaryCertificate: CwaCovidCertificate? = null,
+    colorShade: PersonColorShade,
+    statusBadgeText: Int = 0,
+    badgeCount: Int = 0,
+    onCovPassInfoAction: () -> Unit
+) {
+    val valid = primaryCertificate.isValid
```

Comment:
`isValid` doesn't return `true` for Tests with state `Expiring_soon`, see https://github.com/corona-warn-app/cwa-app-android/pull/4493#discussion_r764766447. (mentioned by @mtwalli )

Therefore, tests that are expiring soon would be displayed as invalid...

I think we have this bug in other places as well, e.g. in `PersonDetailsViewModel`...

Reply:
Actually `isValid` returns `true` for expired/expiring soon test certificates. The only time it returns false for a test certificate is when the state of the certificate is `State.Invalid` (I know, it's confusing). For vaccination and recovery certificates we don't make this exception & I don't remember why we need to display expired test certificates as valid but we had a whole discussion on this when we first introduced the certificates screen.
I remember writing that `isValid` value in the `CwaCovidCertificate` interface with @mtwalli in a pair programming session when we first introduced the certificates screen.

I don't think it should change for the UI here. 
@mtwalli's comment on your PR made sense because we need to have only the actual valid certificates to calculate the admission state, but for the UI, we need to display expired/expiring soon test certificates as valid. 

@mtwalli, you were there with me when we implemented this feature, please give your input here. Am I making sense? 😰

Reply:
@SamuraiKek  with this change , how expired TC is suppose to be displayed ?

Reply:
@SamuraiKek 
I was not involved but I could imagine that `isValid` and `State.Invalid` differ by technical expiration (COSE header timestamps; technical validity) and test expiration (DCC timestamps; effective test validity). It can be useful to keep expired tests (effective validity) as valid (technical validity) as restrictions may differ between EU countries (e.g., RAT test valid 24hrs vs. 48hrs).
Was it this why the destinction was made?

Reply:
> @SamuraiKek with this change , how expired TC is suppose to be displayed ?

An expired TC will still be displayed as valid. 
I haven't changed any logic, I've only introduced a new layout just for the overview screen. Until now, we had the same layout shared between the overview screen QR card and the person details QR card. This new feature that lets you select different certs on the overview screen would have introduced even more bloat to the shared UI, that's why I did the separation.

In terms of behaviour, it's the same as before, only now we have these two additional buttons in the case of 2G/2G+ certs.

Reply:
> @SamuraiKek I was not involved but I could imagine that `isValid` and `State.Invalid` differ by technical expiration (COSE header timestamps; technical validity) and test expiration (DCC timestamps; effective test validity). It can be useful to keep expired tests (effective validity) as valid (technical validity) as restrictions may differ between EU countries (e.g., RAT test valid 24hrs vs. 48hrs). Was it this why the destinction was made?
> 
> Edit: Wasn't `expiring_soon` used with regard to technical validity (to notify user to get a new (vaccination) DCC issued)?

@vaubaehn, this might be the case, and yes, `expiring_soon` is used mostly to notify the user to get a new vaccine.

Reply:
@SamuraiKek , @LukasLechnerDev , @vaubaehn , we have differentiate between two things:
1- How certificates are displayed in the App?
2- What certificates are eligible for G events? 

Lukas did work on determining the state based on VALID certificates (Valid,Expiring_Soon) which is correct ,since expired TC should not be considered in the admission state.

Alexandru is working on how to display the certificates in the cards (UI) , Expired TC should be displayed as Valid (without the whitish overlay).

I think `isValid` flag was used mainly for this purpose only, I suggest to rename it to `isDisplayValid` 

@SamuraiKek  Please make sure that Expired TC has to displayed as pre this change ?

Reply:
@mtwalli, @LukasLechnerDev, I've renamed `isValid` to `isDisplayValid` and rechecked that the expired test certificates are displayed as valid. All comments should be addressed now.